trustmanager/keyfilestore.go

```diff
@@ -147,7 +316,44 @@ func (s *KeyMemoryStore) ExportKey(name string) ([]byte, error) {
 // ImportKey imports the private key in the encrypted bytes into the keystore
 // with the given key ID and alias.
 func (s *KeyMemoryStore) ImportKey(pemBytes []byte, alias string) error {
-	return importKey(s, s.Retriever, s.cachedKeys, alias, pemBytes)
+	err := importKey(s, s.Retriever, s.cachedKeys, alias, pemBytes)
+	if err != nil {
+		return err
+	}
+	keyID, keyInfo := getImportedKeyInfo(pemBytes, alias)
+	s.keyInfoMap[keyID] = keyInfo
+	return nil
+}
+
+func getImportedKeyInfo(pemBytes []byte, filename string) (string, KeyInfo) {
+	var keyID, gun, role string
+	// Try to read the role and gun from the filepath and PEM headers
+	if filepath.HasPrefix(filename, "root_keys") {
```

Comment:
definitely agreed on using the notary consts, I'll also see what I can do with `generateKeyInfoMap`